src/plugins/Logging.ts

```diff
@@ -1,67 +1,80 @@
+import { LOG_LEVELS } from '../types';
+import dispatchEvent from '../utils/dispatchEvent';
+import onEvent from '../utils/onEvent';
 import GenericPlugin from './GenericPlugin';
 
 const BASE_MESSAGE = '[DEBUG]';
 
-let adId = 0;
-
 class LoggingPlugin extends GenericPlugin {
   public onCreate() {
-    const id = ++adId;
-    (window as any)[`ad${id}`] = this.ad;
+    const { ad } = this;
+    const { id } = ad;
+
+    (window as any)[`ad${id}`] = ad;
 
     console.log(
       BASE_MESSAGE,
       'Ad Instantiated and assigned as',
-      `window.ad${adId}`,
+      `window.ad${id}`,
     );
+
+    onEvent(id, (event) => {
+      const { level, creator, message, data } = event.detail;
+
+      // @ts-ignore // TODO: index this
+      console[level](
+        `[${level.toUpperCase()}] ${creator} ${message} ${data || ''}`,
```

Comment:
This will read as `[WARN] LoggingPlugin 'Ad has done whatever.'` and will have data if it exists. We might want it stringified.